flink-rpc/flink-rpc-akka/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -413,7 +413,7 @@ static Object deserializeValueIfNeeded(Object o, Method method) {
         if (o instanceof AkkaRpcSerializedValue) {
             try {
                 return ((AkkaRpcSerializedValue) o)
-                        .deserializeValue(AkkaInvocationHandler.class.getClassLoader());
+                        .deserializeValue(Thread.currentThread().getContextClassLoader());
```

Comment:
I'd suggest to use the `flinkClassLoader` instead of the context class loader. I think this is a bit more robust and explicit.

Reply:
of course, good catch